test/test_xbase64.cpp

```diff
@@ -6,11 +6,12 @@
 * The full license is in the file LICENSE, distributed with this software. *
 ****************************************************************************/
 
-#include "gtest/gtest.h"
+#include "xtl/xbase64.hpp"
```

Comment:
Why inverting include orders? Is that required for building when exceptions are disabled?

Reply:
I changed the include order in the test so that the  tested .h file was included first. This flushes out any missing dependencies from the .h file when building, in the spirit of:

https://include-what-you-use.org/

The issue with include-what-you-use problems is that if you attempt to use the .h file in a .cc file then it might fail to build.

This flushed out several missing #include directives in a few .h files.